File: 1635-number-of-good-pairs/1635-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
      HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
      for(int i=0; i<nums.length; i++){
        int count = map.getOrDefault(nums[i], 0);
        map.put(nums[i], ++count);
      }
      int pairs = 0;
      for(Map.Entry<Integer, Integer> entry : map.entrySet()){
        int count = entry.getValue();
        if(count > 1) pairs+= ((count*(count-1))/2);
      }
      return pairs;
    }
}

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    const map = {};
    for(let i=0; i<nums.length; i++){
      if(!map[nums[i]]) map[nums[i]] = 0;
      map[nums[i]]++;
    }

    let pairs = 0;

    for(const key in map){
      let count = map[key];
      pairs+=( count * (count-1) / 2);
    }
    return pairs;
};